### ISSUE TITLE
bug: WS stops working on iOS safari after some time and no reconnect happens

### ISSUE THREAD

**Issue by quolpr** [trpc/trpc#4078]
### Provide environment information

```
  System:
    OS: macOS 13.0
    CPU: (10) arm64 Apple M1 Max
    Memory: 7.35 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.16.0 - ~/.asdf/installs/nodejs/16.16.0/bin/node
    Yarn: 1.22.19 - ~/.asdf/installs/nodejs/16.16.0/.npm/bin/yarn
    npm: 8.11.0 - ~/.asdf/plugins/nodejs/shims/npm
    Watchman: 2023.01.23.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 111.0.5563.110
    Firefox: 109.0.1
    Safari: 16.1
    Safari Technology Preview: 16.4
```

### Describe the bug

After some time when tab is suspended on iOS safari, WS stops working on iOS safari. WebSocket doesn't emit `close` event, and it just hangs. No reconnect happen

### To reproduce

1. Open site with iOS Safari
2. Switch to other apps
3. Wait ~3-5 min
4. Return to safari browser
5. WS stops working, no reconnect happens

I also checked the "network" tab. The connection looks like is not closed, but not events are emitting.

### Additional information

I found a dirty workaround:

```ts
document.addEventListener("visibilitychange", () => {
  // Let's reconnect on visibility change,
  // Cause, for example, iOS safari connection may hang
  if (document.visibilityState === "visible") {
    wsClient.getConnection().dispatchEvent(new CloseEvent("close"));
  }
});
```

And, an interesting thing, is that I need to make `wsClient.getConnection().dispatchEvent(new CloseEvent("close"))`. If I do just `wsClient.close()` — nothing happens. I think `close` event just not emitting when WebSocket is hanged.

I found the same issue at `graphql-ws`, so it may help https://github.com/enisdenjo/graphql-ws/discussions/290 . They recommend using this solution — https://the-guild.dev/graphql/ws/recipes#client-usage-with-abrupt-termination-on-pong-timeout

<sub>[TRP-16](https://linear.app/trpc/issue/TRP-16/bug-ws-stops-working-on-ios-safari-after-some-time-and-no-reconnect)</sub>

## Funding
* You can sponsor this specific effort via a [Polar.sh](https://polar.sh) pledge below
* We receive the pledge once the issue is completed & verified

<a href="https://polar.sh/trpc/trpc/issues/4078">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/trpc/trpc/issues/4078/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/trpc/trpc/issues/4078/pledge.svg">
</picture>
</a>

**Comment by dieferomanoski** [comment]
Did you find any workaround for this?

**Comment by quolpr** [comment]
@dieferomanoski check "I found a dirty workaround:" section in my first message 🙂

**Comment by KATT** [comment]
I think this is solved in the latest `@next` or earlier, feel free to reopen otherwise

**Comment by Azarattum** [comment]
What is the solution, exactly? Do you reconnect on visibility change like it has been suggested in the workaround?

**Comment by KATT** [comment]
I implemented `keepAlive` logic, will be merged soon, see #6030 

Further improvements to the docs is welcome
